helm/distributed-idol/templates/dih/statefulset.yaml

```diff
@@ -34,116 +26,32 @@ spec:
-        
-        lifecycle:
-          preStop:
-            exec:
-              command:
-              - /bin/bash
-              - "-c"
-              - |
-{{ tpl (.Files.Get "resources/dih_preStop.sh") . | indent 16 }}
-        volumeMounts:
-        - name: dih-persistent-storage
-          mountPath: /opt/idol/dih/data
-        {{- range $component.additionalVolumeMounts }}
-        - {{ . | toYaml | nindent 10 }}
-        {{- end }}
-        env:
-        - name: IDOL_COMPONENT_CFG
-          # Override default in newer IDOL containers
-          value: "/dih/dih.cfg"
-        {{- if $component.envConfigMap }}
-        envFrom:
-        - configMapRef: {{ $component.envConfigMap | quote }}
-        {{ end }}
-        livenessProbe:
-          httpGet:
-            path: /a=getpid
-            port: {{ $component.aciPort | int }}
-{{- template "idol-library.standardLivenessProbe" $component.livenessProbe }}
-        ports:
-        - containerPort: {{ $component.aciPort | int }}
-          name: aci-port
-          protocol: TCP
-        - containerPort: {{ $component.servicePort | int }}
-          name: service-port
-          protocol: TCP
-        - containerPort: {{ $component.indexPort | int }}
-          name: index-port
-          protocol: TCP
-{{- /*
- Exporter monitors child fullness - unused in mirrored setup 
-*/}}
-{{- if and (.Values.autoscaling.enabled) (not .Values.setupMirrored) }}
-      - name: prometheus-exporter
-        image: python:3.11
-{{- if (coalesce .Values.global.http_proxy .Values.global.https_proxy) }}
-        envFrom:
-          - configMapRef:
-              name: http-proxy-config
+{{- if (dig "podSecurityContext" "enabled" false ($component | merge (dict))) }}
+      securityContext: {{- omit $component.podSecurityContext "enabled" | toYaml | nindent 8 }}
 {{- end }}
-        imagePullPolicy: IfNotPresent
-        workingDir: /usr/src/app
-        command: ["/bin/sh"]
-        args: ["-c", "mkdir -p /usr/src/app && 
-          echo \"Copying files\" && 
-          cp /mnt/python/* /usr/src/app && 
-          cd /usr/src/app && 
-          echo \"Installing\" &&
-          pip install --no-cache-dir -r requirements.txt &&
-          echo \"Running\" &&
-          python dih_prometheus_exporter.py"]
-        volumeMounts:
-        - name: dih-prometheus-exporter-python
-          mountPath: /mnt/python
-        - name: python-src-app
-          mountPath: /usr/src/app
-        - name: python-local
-          mountPath: /.local
-        ports:
-        - containerPort: {{ $component.prometheusPort| int }}
-          name: metrics-port
-          protocol: TCP
+{{- if (dig "resources" "enabled" false ($component | merge (dict))) }}
+        resources: {{- omit $component.resources "enabled" | toYaml | nindent 10 }}
 {{- end }}
       volumes:
       - name: config-map
         configMap:
           name: {{ default (printf "%s-default-cfg" $component.name) $component.existingConfigMap }}
```

Comment:
ideally volumes should use the template arg so we don't have to duplicate this from idol-library

Reply:
https://github.com/opentext-idol/idol-containers-toolkit/pull/80/commits/a7e2b2a4a97437f61ca3175da86b172b9cf136b1